frontend/src/scenes/organization/Settings/inviteLogic.ts

```diff
@@ -0,0 +1,125 @@
+import { kea } from 'kea'
+import { OrganizationInviteType } from '~/types'
+import api from 'lib/api'
+import { toast } from 'react-toastify'
+import { organizationLogic } from 'scenes/organizationLogic'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { inviteLogicType } from './inviteLogicType'
+import { preflightLogic } from 'scenes/PreflightCheck/logic'
+import { successToast } from 'lib/utils'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+
+/** State of a single invite row (with input data) in bulk invite creation. */
+interface InviteRowState {
+    target_email: string
+    first_name: string
+    isValid: boolean
+}
+
+const EMPTY_INVITE: InviteRowState = { target_email: '', first_name: '', isValid: true }
+
+export const inviteLogic = kea<inviteLogicType<InviteRowState>>({
+    path: ['scenes', 'organization', 'Settings', 'inviteLogic'],
+    actions: {
+        showInviteModal: true,
+        hideInviteModal: true,
+        updateInviteAtIndex: (payload, index: number) => ({ payload, index }),
+        deleteInviteAtIndex: (index: number) => ({ index }),
+        appendInviteRow: true,
+        resetInviteRows: true,
+    },
+    connect: {
+        values: [preflightLogic, ['preflight']],
+    },
+    reducers: () => ({
+        isInviteModalShown: [
+            false,
+            {
+                showInviteModal: () => true,
+                hideInviteModal: () => false,
+                inviteTeamMembersSuccess: () => false,
+                [router.actionTypes.locationChanged]: () => false,
```

Comment:
Nice trick! The only issue may come when we get a page that decides to change its url somehow, but that's for our future colleagues to fix. 🤞